lightningd/opening_control.c

```diff
@@ -305,12 +303,12 @@ static void opening_funder_start_replied(struct subd *openingd, const u8 *resp,
 					 const int *fds,
 					 struct funding_channel *fc)
 {
-	u8 *funding_scriptPubkey;
 	bool supports_shutdown_script;
 
-	if (!fromwire_openingd_funder_start_reply(resp, resp,
-						 &funding_scriptPubkey,
-						 &supports_shutdown_script)) {
+	if (!fromwire_openingd_funder_start_reply(fc, resp,
+						  &fc->funding_scriptpubkey,
+						  &supports_shutdown_script,
```

Comment:
I take it that the upfront shutdown script is negotiated independently of the channel_type? Or could it be removed and passed via the channel_type instead?